src/http/ContentLengthInterpreter.h

```diff
@@ -19,12 +21,21 @@ namespace Http
 class ContentLengthInterpreter
 {
 public:
-    explicit ContentLengthInterpreter(const int aDebugLevel);
+    ContentLengthInterpreter();
 
     /// updates history based on the given message-header field
     /// \return true iff the field should be added/remembered for future use
     bool checkField(const String &field);
 
+    /// prohibits Content-Length in 1xx and 204 responses
+    void applyStatusCodeRules(const StatusCode code) { prohibitedAndIgnored = ProhibitsContentLength(code); }
+    // TODO: implement
+    /// prohibits Content-Length in GET/HEAD requests
+    // void applyRequestMethodRules(const Http::MethodType method);
+
+    /// prohibits Content-Length in trailer
+    void applyTrailerRules() { prohibitedAndIgnored = true; }
```

Comment:
I would change boolean prohibitedAndIgnored field type to a c-string so that this and other methods can set the reason why the content length is prohibited and ignored (and the ignoring code can report that reason). I do not insist on this change.

Reply:
Done.